documentation/development-docs/design-guidelines/should-process-confirm-impact.md

```diff
@@ -1,10 +1,12 @@
 ## ShouldProcess
 
-From [Cmdlet Attribute Declaration](https://msdn.microsoft.com/en-us/library/system.management.automation.cmdletcommonmetadataattribute.supportsshouldprocess(v=vs.85).aspx):
+From [_Cmdlet Attribute Declaration_](https://docs.microsoft.com/en-us/powershell/developer/cmdlet/cmdlet-attribute-declaration):
 
-> _All commands that change the system should include the **SupportsShouldProcess** keyword when declaring the attribute. If the ShouldProcess method returns true, the command should perform its action. If the ShouldProcess method returns false, the action should not be taken._
-> 
-> _If a command supports the ShouldProcess method, the **Confirm** and **WhatIf** parameters are added to the command by the Windows PowerShell runtime._
+> _All cmdlets that change resources outside of PowerShell should include the `SupportsShouldProcess` keyword when the `Cmdlet` attribute is declared, which allows the cmdlet to call the `ShouldProcess` method before the cmdlet performs its action. If the `ShouldProcess` call returns false, the action should not be taken._
+>
+> _The `-Confirm` and `-WhatIf` cmdlet parameters are available only for cmdlets that support `ShouldProcess` calls._
+
+Below is an example of the declaration of this property in the `Cmdlet` attribute and the subsequent call to `ShouldProcess` in the body of the cmdlet:
 
 ```cs
 [Cmdlet(VerbsCommon.Remove, "Foo", SupportsShouldProcess = true)]
```

Comment:
we could also link an [example](https://github.com/Azure/azure-powershell/pull/8755/files#diff-3d8a9105c54f1a4fa126c6c97b52be44R68) from an `*exmaple` file and give a forward declaration of the Helpers: `ConfirmAction` and may be add an example the the helpers.

Reply:
Added a link out to the `examples` folder with these sample implementations.